proposals/low-level-struct-improvements.md

```diff
@@ -995,6 +1106,53 @@ struct Dimensions
 
 ## Future Considerations
 
+### Allowing attributes on locals
+Another friction point for developers using `ref struct` is local variables 
+can suffer from the same issues as parameters with respect to their lifetimes 
+being decided at declaration. Than can make it difficult to work with 
+`ref struct` that are assigned on multiple paths where at least one of the 
+paths is a limited *safe-to-escape* scope. 
+
+```cs
+int length = ...;
+Span<byte> span;
+if (length > StackAllocLimit)
+{
+    span = new Span(new byte[length]);
+}
+else
+{
+    // Error: The *safe-to-escape* of `span` was decided to be outside the 
+    // current method scope hence it can't be the target of a stackalloc
+    span = stackalloc byte[length];
+}
+```
+
+For `Span<T>` specifically developers can work around this by initializing the 
+local with a `stackalloc` of size zero. This changes the *safe-to-escape* scope
+to be the current method and is optimized away by the compiler. It's effectively
+a syntax for making a `[DoesNotEscape]` local.
+
+```cs
+int length = ...;
+Span<byte> span = stackalloc byte[0];
+if (length > StackAllocLimit)
+{
+    span = new Span(new byte[length]);
+}
+else
+{
+    // Okay
+    span = stackalloc byte[length];
+}
+```
+
+This only works for `Span<T>` though, there is no general purpose mechanism for
+`ref struct` values. However the `[DoesNotEscape]` attribute provides exactly 
+the semantics that are desired here. If we decide in the future to allow 
+attributes to apply to local functions it would provide immediate relief to this
```

Comment:
We allow attributes to apply to local functions.

Reply:
Blah, I meant to say local variables .